network-core/src/subscription.rs

```diff
@@ -0,0 +1,25 @@
+use chain_core::property::{Block, HasHeader};
+
+use std::fmt::{self, Debug};
+
+pub enum BlockEvent<B>
+where
+    B: Block + HasHeader,
+{
+    Announce(<B as HasHeader>::Header),
+    Solicit(Vec<<B as Block>::Id>),
+}
+
+impl<B> Debug for BlockEvent<B>
+where
+    B: Block + HasHeader,
+    <B as HasHeader>::Header: Debug,
+    <B as Block>::Id: Debug,
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            BlockEvent::Announce(header) => f.debug_tuple("Announce").field(header).finish(),
+            BlockEvent::Solicit(ids) => f.debug_tuple("Solicit").field(ids).finish(),
```

Comment:
nice, I never really looked into it before (the `debug` feature)

Reply:
It's what `#[derive(Debug)]` does automatically, but here it needs to be done manually because of how the type parameter is not part of the contents, but its associated types are.